ale/transformation.py

```diff
@@ -137,3 +164,31 @@ def rotation_to(self, other):
         for next_rotation in rotations[1:]:
             rotation = next_rotation * rotation
         return rotation
+
+    def last_time_dependent_frame_between(self, other):
+        """
+        Find the last time dependent frame between this frame and another frame.
+
+        Parameters
+        ----------
+        other : FrameNode
+            The frame to find the last time dependent frame between
+
+        Returns
+        -------
+        FrameNode
+            The first frame between the this frame and the other frame such
+            that the rotation from this frame to the in-between frame is time
+            dependent and the rotation from the in-between frame to the other
+            frame is constant. If there are no time dependent frames between
+            this frame and the other frame, None is returned.
+        """
```

Comment:
I don't know a clean way to name or describe this method. It's a rather complex thing.

Reply:
I have not reviewed this or down - have to step away for a bit